_source/_posts/2022-08-08-react-redux.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: blog_post
+title: Using Redux with React
+author: gabi-dombrowski
+by: contractor
+communities: [javascript]
+description: ""
+tags: [javascript, typescript, react, redux, usecontext, usestate, authentication]
+tweets:
+  - ""
+  - ""
+  - ""
+image:
+type: conversion
+canonical: https://auth0.com/blog/using-redux-with-react/
+---
+
+There are a myriad of state management options available for React. React provides the option of using the built-in [Context](https://reactjs.org/docs/context.html) for when you have a nested tree of components that share a state. There is also a built-in [useState hook](https://reactjs.org/docs/context.html) that will allow you to set local state for a component.
+
+For more complex scenarios where you might need a single source of truth that shared across large sections of your application and is frequently changing, you might want to consider using a more robust state management library.
+
+Let's compare when to use the useState hook vs. React Context vs. a more global state management solution like one of the most popular ones, [Redux](https://redux.js.org/).
+
+## Initial Setup
+
+We'll get started with the built-in React template for [Create React App](https://github.com/facebook/create-react-app) that uses Redux and Typescript. This includes the recommended Redux Toolkit and Redux's integration with React components.
+
+1. First, run:
+```bash
+npx create-react-app <YOUR_APP_NAME> --template redux-typescript
```

Comment:
```suggestion
npx create-react-app@5.0 okta-react-redux --template redux-typescript
```

Suggested name, but feel free to go with something else. IMO it'll be easier for people to have a suggested name.

Also, I try to add versions to all npm (and npx) commands. Does adding the version here make sense or do you disagree with specifying a version for `create-react-app` (there might be reasons against this in the community, so getting your feel on it).

Reply:
Considering the simplicity of the app, I would prefer the tutorial stand up an app that is whatever latest stable version is available.

The only dependencies are node and npm versioning, which I'm good with saying something like:

```As of the time of this publication, Node >= 14.0.0 and npm >= 5.6 are required for Create React App. You can check  for the latest required versions at https://reactjs.org/docs/create-a-new-react-app.html.```

Think that sounds good?

Reply:
I'm good with adding info to prereqs. If you can also state the version of React that it created (same with all the other deps added) that will help set expectations.

The problem we've seen is that while the app itself is simple, there's compatibility issues with the Okta libs with new versions of different frameworks. We've had comments on posts where people run into problems because of it. We can try without a version here (as long as there's some statement about which version of React/create-react-app this all was tested on) and address any comments should they arise. 

Does that sound ok to you?

Reply:
Yup, that works!